### PR TITLE
feat(as-3017): store file metadata in blob storage rather than cosmos db

### PR DESCRIPTION
**Pull request by shepda**
## Ticket Number
https://pins-ds.atlassian.net/browse/AS-3017

## Description of change
Store the file metadata in Blob Storage as attributes of the uploaded file instead of in Cosmos DB.

Updated and added new unit tests for the changed files (for which the coverage is at 100%) but not added missing unit tests for unchanged files, which is something that should be done at a later date.

This functionality will work for all new files uploaded and the endpoint to migrate the existing data from Cosmos DB to Blob Storage will be done in a different ticket.

Have not removed the mongo functionality as it'll be needed for the migration endpoint and this can be removed in a future ticket with the migration endpoint once the migration has been done. 

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by swdpcomputing — CHANGES_REQUESTED** [review]

File: packages/document-service-api/src/controllers/documents.js [hunk 1]
```diff
@@ -1,237 +1,190 @@
...
-      applicationId: req.params.applicationId,
-    });
-
-    if (docs.length === 0) {
-      req.log.debug('Unknown application ID');
-
-      res.status(404).send({
-        message: 'Unknown application ID',
-      });
+  initContainerClient,
```
[the remark this answers is not in the document]
Comment (shepda) [reply packages/document-service-api/src/controllers/documents.js:2]: As discussed, these could be moved into a common function but with the number of params that would need to be passed to the common function it's not necessarily going to make the parent code any shorter.

File: packages/document-service-api/src/controllers/documents.js [hunk 2]
```diff
@@ -1,237 +1,190 @@
...
       return;
     }
 
-    req.log.info('Downloading file from blob storage');
-
-    const fileBuffer = await doc.downloadFileBuffer();
+    req.log.info('Downloading file');
 
+    const fileBuffer = await downloadFile(containerClient, metadata.location);
     const displayBase64 = req.query.base64 === 'true';
```
[the remark this answers is not in the document]
Comment (shepda) [reply packages/document-service-api/src/controllers/documents.js]: Yep fair enough, I've changed it

File: packages/document-service-api/src/lib/blobStorage.spec.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        await getMetadataForAllFiles(containerClient, applicationId);
+        throw new Error('Expected error not thrown');
+      } catch (err) {
+        expect(err.message).toEqual('Internal Server Error');
+      }
+    });
+  });
+
+  describe('getMetadataForSingleFile', () => {
+    it('should return true when the metadata is fetched successfully', async () => {
```
[the remark this answers is not in the document]
Comment (shepda) [reply packages/document-service-api/src/lib/blobStorage.spec.js]: Well spotted, I've changed it

**Review by rcmsoftwaredevelopment — CHANGES_REQUESTED** [review]
1) Can we have some documentation for other devs/testers on how this can be tested locally and where to look in the Azure blob storage.
2) Can the docker file under the root directory be amended so for the document storage api removes all dependancies to mongodb.

**Review by swdpcomputing — DISMISSED** [review]

**Comment by shepda** [comment]
>     1. Can we have some documentation for other devs/testers on how this can be tested locally and where to look in the Azure blob storage.

@rcmsoftwaredevelopment Yes sure will add some docs

**Comment by shepda** [comment]
>     2. Can the docker file under the root directory be amended so for the document storage api removes all dependancies to mongodb.

@rcmsoftwaredevelopment We'll need to leave the mongo connectivity in for now as we'll need it for the migration tool and we can remove it with the migration tool once the migration has been done in prod.

**Review by rcmsoftwaredevelopment — APPROVED** [review]

**Pull request merged**